x-pack/plugins/apm/ftr_e2e/cypress/e2e/power_user/onboarding/onboarding.cy.ts

```diff
@@ -0,0 +1,158 @@
+      cy.contains("gem 'elastic-apm'");
+
+      cy.contains('Go').click();
+      cy.contains('go get go.elastic.co/apm');
+
+      cy.contains('Java').click();
+      cy.contains('-javaagent');
+
+      cy.contains('.NET').click();
+      cy.contains('Elastic.Apm.NetCoreAll');
+
+      cy.contains('PHP').click();
+      cy.contains('apk add --allow-untrusted <package-file>.apk');
+
+      cy.contains('OpenTelemetry').click();
+      cy.contains('Download the OpenTelemetry APM Agent or SDK');
+    });
+  });
+
+  describe('check Agent Status', () => {
+    beforeEach(() => {
+      cy.loginAsEditorUser();
+      cy.visitKibana('/app/apm/onboarding');
+    });
+    it('when no data is present', () => {
+      cy.intercept('GET', '/internal/apm/agent_status').as('agentStatus');
+      cy.getByTestSubj('checkAgentStatus').click();
+      cy.wait('@agentStatus');
+      cy.getByTestSubj('agentStatusWarningCallout');
+    });
+
+    it('when data is present', () => {
+      synthtrace.index(
+        generateData({
+          from: new Date(start).getTime(),
+          to: new Date(end).getTime(),
+        })
+      );
+      cy.intercept('GET', '/internal/apm/agent_status').as('agentStatus');
+      cy.getByTestSubj('checkAgentStatus').click();
+      cy.wait('@agentStatus');
+      cy.getByTestSubj('agentStatusSuccessCallout');
+      synthtrace.clean();
+    });
+  });
+
+  describe('create API Key', () => {
+    it('create the key successfully', () => {
+      cy.loginAsApmManageOwnAndCreateAgentKeys();
+      cy.visitKibana('/app/apm/onboarding');
+      cy.intercept('POST', '/api/apm/agent_keys').as('createApiKey');
+      cy.getByTestSubj('createApiKeyAndId').click();
+      cy.wait('@createApiKey');
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Django').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Flask').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Ruby on Rails').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Rack').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Go').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('Java').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('.NET').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('PHP').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+
+      cy.contains('OpenTelemetry').click();
+      cy.getByTestSubj('apiKeySuccessCallout');
+    });
+
+    it('fails to create the key due to missing privileges', () => {
+      cy.loginAsEditorUser();
+      cy.visitKibana('/app/apm/onboarding');
+      cy.intercept('POST', '/api/apm/agent_keys').as('createApiKey');
+      cy.getByTestSubj('createApiKeyAndId').click();
+      cy.wait('@createApiKey');
+      cy.getByTestSubj('apiKeyWarningCallout');
+    });
+  });
+});
```

Comment:
I general, I think this test should have clear expectations.

Reply:
I will also add the check for intercepted network request status. 
Failed to create Api Key only results in 2 things, 1 is a warning callout and 2nd is that the request failed with 403